micro-rdk/src/common/data_manager.rs

```diff
@@ -20,6 +24,37 @@ pub enum DataManagerError {
     StoreError(#[from] DataStoreError),
     #[error("queried time interval {0} is not a multiple of minimum time_interval{1}")]
     ImproperTimeInterval(u64, u64),
+    #[error("data service config does not exist or is improperly configured")]
+    ConfigError,
+    #[error(transparent)]
+    InitializationRobotError(#[from] RobotError),
+}
+
+fn get_data_sync_interval(cfg: &ConfigResponse) -> Result<Option<Duration>, DataManagerError> {
+    let robot_config = cfg.config.clone().ok_or(DataManagerError::ConfigError)?;
+    Ok(
+        if let Some(data_cfg) = robot_config
+            .services
+            .iter()
+            .find(|svc_cfg| svc_cfg.r#type == *"data_manager")
```

Comment:
What if there is more than one in the config?

Reply:
I don't think this is an expected occurrence, though possible. Now that you mention, I'm just going to throw a special error if there are multiple